src/__test__/mocks/getWindow.ts

```diff
@@ -0,0 +1,51 @@
+/*
+  Load a jsdom.DOMWindow
+*/
+// tslint:disable:no-expression-statement
+import * as request from 'supertest'
+import { JSDOM, DOMWindow, ResourceLoader } from 'jsdom'
+
+
+const fakeUrl = 'https://test-roar.morehumaninternet.org'
+
+function loadLinkOrScript(el: HTMLLinkElement | HTMLScriptElement): Promise<void> {
+  const timeoutError = new Error('timeout')
+  return new Promise((resolve, reject) => {
+    const timeout = setTimeout(() => reject(timeoutError), 5000)
+    el.addEventListener('load', () => {
+      clearTimeout(timeout)
+      resolve()
+    })
+  })
+}
+
+function loadAllLinksAndScripts(window: DOMWindow): Promise<any> {
+  const links: ReadonlyArray<HTMLLinkElement | HTMLScriptElement> = Array.from(window.document.getElementsByTagName('link'))
+  const scripts = Array.from(window.document.getElementsByTagName('script'))
+  const toLoad = links.concat(scripts)
+  return Promise.all(toLoad.map(loadLinkOrScript))
+}
+
+export async function getWindow(agent: request.SuperTest<request.Test>, path: string): Promise<DOMWindow> {
+  const response = await agent.get(path)
+
+  const { window } = new JSDOM(response.text, {
+    url: fakeUrl,
+    pretendToBeVisual: true,
+    runScripts: 'dangerously',
+    resources: Object.assign(new ResourceLoader(), {
```

Comment:
I read [this](https://github.com/jsdom/jsdom#loading-subresources) section from JSDom docs and I don't understand why we can use `resources: "usable"`. Can you please explain?

Reply:
We only want to fetch specific resources, namely the ones from our own server. Perhaps we want to do this over the network, but using the `agent` worked